src/nvim/testdir/test_eval_stuff.vim

```diff
@@ -11,3 +11,14 @@ endfunction
 func Test_catch_return_with_error()
   call assert_equal(1, s:foo())
 endfunc
+
+func Test_nocatch_restore_silent_emsg()
+  silent! try
+    throw 1
+  catch
+  endtry
+  redir @z
+  echoerr 'wrong'
+  redir END
```

Comment:
`emsg_silent` only affects display, not redirect, so this test isn't valid unfortunately.